### PR TITLE
Archer v2 bug

### PR DESCRIPTION
**Pull request by merit-maita**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/37752

## Description
added a parameter as an api endpoint for archer v2.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2.yml [hunk 1]
```diff
@@ -66,6 +66,12 @@ configuration:
   name: proxy
   required: false
   type: 8
+- display: API Endpoint
+  hidden: false
+  name: ApiEndpoint
```
Comment (bakatzir) [inline Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2.yml:71]: name field can be snakified, to follow python conventions.

File: Packs/ArcherRSA/ReleaseNotes/1_1_19.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### RSA Archer v2
+- Added an API Endpoint as a parameter, with the name API Endpoint.
```
Comment (bakatzir) [inline Packs/ArcherRSA/ReleaseNotes/1_1_19.md:4]:
```suggestion
- Fixed an issue of hard coded API endpoints in the code by adding a new *API Endpoint* parameter to the integration.
```

File: Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2.yml [hunk 3]
```diff
@@ -66,6 +66,12 @@ configuration:
   name: proxy
   required: false
   type: 8
+- display: API Endpoint
+  hidden: false
+  name: api_endpoint
+  required: false
+  defaultvalue: api
+  type: 0
```
Comment (yaakovi) [inline Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2.yml:74]: * move it up to be right after the `Server URL` parameter (as they belong together).
* change the default value to `rsaarcher/api`
* in the code, make sure it doesn't create a duplicate in the URL (`<base URL>/rsaarcher/rsaarcher/api`)
* remove the `rsaarcher` suffix from the `Server URL` example.

File: Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2.py [hunk 4]
```diff
@@ -32,6 +32,8 @@
 
 ACCOUNT_STATUS_DICT = {1: 'Active', 2: 'Inactive', 3: 'Locked'}
 
+API_ENDPOINT = demisto.params().get('api_endpoint', 'api')
```
Comment (yaakovi) [inline Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2.py:35]: * move to the main.
* use the following implementation of default value:
`params.get('api_endpoint') or 'rsaarcher/api'`

File: Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2.py [hunk 5]
```diff
@@ -288,7 +290,7 @@ def update_session(self):
             'Password': self.password
         }
         try:
-            res = self._http_request('POST', '/api/core/security/login', json_data=body)
+            res = self._http_request('POST', f'/{API_ENDPOINT}/core/security/login', json_data=body)
```
Comment (yaakovi) [inline Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2.py:293]: add the API endpoint to the base URL in the main function instead of applying it each time

**Review by yaakovi — CHANGES_REQUESTED** [review]

**Review by yaakovi — APPROVED** [review]

**Pull request merged**